app/services/metrics_recorder.rb

```diff
@@ -1,8 +1,9 @@
 require 'prometheus/client'
-require 'singleton'
 
 class MetricsRecorder
-  include Singleton
+  def self.instance
```

Comment:
Instead of lazy-instantiation, use a class variable that is immediately initialized and return that in `instance`.